.github/workflows/ubuntu.yml

```diff
@@ -0,0 +1,37 @@
+name: Build on ubuntu
+
+on:
+  push:
+  workflow_dispatch:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        submodules: true
+    - run: |
+        git config --global --add safe.directory '*'
```

Comment:
can you please explain why do we need this config ?

Reply:
I don't recall setting this configuration on my machine. it doesn't show up when I run:
```bash
git config --list
```